shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -137,14 +147,22 @@ void syncUpdatesFromManager() {
                                                 failure -> failedToSendUpdateToManager(y, failure));
                             }
                             if (ManagedResourceStatus.DEPROVISION.equals(y.getStatus())) { // Processor to delete
-                                y.setStatus(ManagedResourceStatus.DELETING);
-                                return notifyProcessorStatusChange(y)
-                                        .subscribe().with(
-                                                success -> {
-                                                    LOGGER.debug("Deleting notification for Processor '{}' has been sent to the manager successfully", y.getId());
-                                                    bridgeExecutorService.deleteBridgeExecutor(y);
-                                                },
-                                                failure -> failedToSendUpdateToManager(y, failure));
+                                if (Objects.isNull(bridgeExecutorService.getBridgeExecutor(y))) {
+                                    LOGGER.debug("Processor '{}' was not found. Notifying manager that it has been deleted.", y.getId());
+                                    y.setStatus(ManagedResourceStatus.DELETED);
+                                    return notifyProcessorStatusChange(y).subscribe().with(
+                                            success -> LOGGER.debug("Deleted notification for Processor '{}' has been sent to the manager successfully", y.getId()),
+                                            failure -> failedToSendUpdateToManager(y, failure));
+                                } else {
+                                    y.setStatus(ManagedResourceStatus.DELETING);
+                                    return notifyProcessorStatusChange(y)
+                                            .subscribe().with(
+                                                    success -> {
+                                                        LOGGER.debug("Deleting notification for Processor '{}' has been sent to the manager successfully", y.getId());
+                                                        bridgeExecutorService.deleteBridgeExecutor(y);
+                                                    },
+                                                    failure -> failedToSendUpdateToManager(y, failure));
+                                }
```

Comment:
I'd rather put this logic in the `BridgeExecutorService` i.e. turn 
```java
    @Override
    public void deleteBridgeExecutor(ProcessorDTO processorDTO) {
        final String namespace = customerNamespaceProvider.resolveName(processorDTO.getCustomerId());
        final boolean bridgeDeleted =
                kubernetesClient
                        .resources(BridgeExecutor.class)
                        .inNamespace(namespace)
                        .delete(BridgeExecutor.fromDTO(processorDTO, namespace, executorImage));
        if (!bridgeDeleted) {
            // TODO: we might need to review this use case and have a manager to look at a queue of objects not deleted and investigate. Unfortunately the API does not give us a reason.
            LOGGER.warn("BridgeExecutor '{}' not deleted", processorDTO);
        }
    }
```
into
```java
    @Override
    public void deleteBridgeExecutor(ProcessorDTO processorDTO) {
        final String namespace = customerNamespaceProvider.resolveName(processorDTO.getCustomerId());
        final boolean bridgeDeleted =
                kubernetesClient
                        .resources(BridgeExecutor.class)
                        .inNamespace(namespace)
                        .delete(BridgeExecutor.fromDTO(processorDTO, namespace, executorImage));
        if (!bridgeDeleted) {
            // TODO: we might need to review this use case and have a manager to look at a queue of objects not deleted and investigate. Unfortunately the API does not give us a reason.
            LOGGER.warn("BridgeExecutor '{}' not deleted", processorDTO);

            // NOTIFY THE MANAGER HERE
        }
    }
```

Reply:
I see your point... but we don't do any notifications to the `manager` from the `BridgeXXXService`'s at present. I think it confusing to notify the `manager` about success from the `ManagerSyncService` but notify it about failures from `BridgeXXXService`. Even the `XXXController`'s notify the `manager` through `ManagerSyncService` and not directly. IMO we should try to be consistent.

That said, this is only my opinion, one of many which clearly differs to yours. Without putting undue pressure on @lucamolteni I will await his review too.. no doubt he will say something different. Once all opinions "are in" I will refactor to some middle ground where we can all be appeased :smiley:

Reply:
@manstis I'd go with the solution with less code, and putting the logic there seems to be a viable solution. Can you please try to see how it works out? Thanks

Reply:
I'd stay with the simplicity. If we can achieve the same goal with ~2 lines instead of ~200 I would go for the first one. Having said that, up to you.

Reply:
Done.